lnwire/fuzz_test.go

```diff
@@ -822,3 +822,24 @@ func FuzzCustomMessage(f *testing.F) {
 		harness(t, data)
 	})
 }
+
+// FuzzParseRawSignature tests that our DER-encoded signature parsing does not
+// panic for arbitrary inputs and that serializing and reparsing the signatures
+// does not mutate them.
+func FuzzParseRawSignature(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte) {
+		sig, err := NewSigFromRawSignature(data)
+		if err != nil {
+			return
+		}
+
+		sig2, err := NewSigFromRawSignature(sig.ToSignatureBytes())
+		if err != nil {
+			t.Fatalf("failed to reparse signature: %v", err)
+		}
+
+		if !reflect.DeepEqual(sig, sig2) {
+			t.Fatalf("signature mismatch: %v != %v", sig, sig2)
+		}
```

Comment:
any reason not to use the `require` package for these? I see that all the other fuzz tests also dont use it... not sure if there was a specific reason for that.

Reply:
I don't have a good reason.  I'll send a follow-up PR making the change for all fuzz tests.